File: app/src/main/java/bible/translationtools/converter/Language.java
package bible.translationtools.converter;

public class Language {
    String slug;
    String name;
    String angName;

    public Language(String slug, String name, String angName) {
        this.slug = slug;
        this.name = name;
        this.angName = angName;
    }

    @Override
    public boolean equals(Object obj) {
        return this.slug.equals(((Language) obj).slug);
    }

    @Override
    public String toString() {
        return String.format("[ %s ] %s", this.slug, this.name) +
                (!this.angName.equals(this.name) && !this.angName.equals("") ? " ( "+ this.angName +" )" : "");
    }
}

File: app/src/main/java/bible/translationtools/converter/Language.kt
package bible.translationtools.converter

class Language(val slug: String, val name: String, val angName: String) {
    override fun equals(other: Any?): Boolean {
        return this.slug == (other as Language).slug
    }

    override fun toString(): String {
        val name = String.format("[ %s ] %s", this.slug, this.name)
        val angName = if (this.angName != this.name && this.angName != "") {
            " ( ${this.angName} )"
        } else ""
        return name + angName
    }

    override fun hashCode(): Int {
        var result = slug.hashCode()
        result = 31 * result + name.hashCode()
        result = 31 * result + angName.hashCode()
        return result
    }
}
